mdb-realm/src/deploy.ts

```diff
@@ -0,0 +1,28 @@
+import * as core from "@actions/core";
+import { createCluster, getConfig } from "./helpers";
+
+async function run(): Promise<void> {
+    try {
+        const config = getConfig();
+        for (let i = 0; i < 5; i++) {
+            try {
+                await createCluster(config);
+                core.setOutput("clusterName", config.clusterName);
+                core.setOutput("atlasUrl", config.atlasUrl);
+                core.setOutput("realmUrl", config.realmUrl);
+
+                return;
+            } catch (e) {
+                core.warning(`An error occurred while deploying cluster '${config.clusterName}': ${e}. Retrying...`);
+            }
+        }
+
+        core.setFailed(`Failed to deploy ${config.clusterName} after 5 attempts.`);
+    } catch (error: any) {
+        core.setFailed(`An unexpected error occurred: ${error.message}\n${error.stack}`);
+    }
+}
+
+run();
```

Comment:
I'd suggest removing your try-catch in the function block and instead capturing any errors thrown when running. This will ensure you also fail correctly in case the the catch block throws.
```suggestion
run().catch(e => {
  const message = error.message || String(error);
  core.setFailed(message);
  core.debug(err.stack || "No stack available");
});
```